muikku-core-plugins/src/main/resources/META-INF/resources/scripts/ckplugins/muikku-organizerfield/plugin.js

```diff
@@ -0,0 +1,413 @@
+    },
+    getTerms: function() {
+      var result = [];
+      var keys = Object.keys(this.terms);
+      for (var i = 0; i < keys.length; i++) {
+        result.push({
+          'id': this.terms[keys[i]].id,
+          'name': keys[i]
+        });
+      }
+      return result;
+    },
+    removeTerm(termId) {
+      var termName = this.getTermName(termId);
+      var termObject = this.terms[termName];
+      if (termObject.count > 1) {
+        termObject.count--;
+      }
+      else {
+        delete this.terms[termName];
+        var index = this.termIds.indexOf(termId);
+        this.termIds.splice(index, 1);
+      }
+    },
+    getCategories: function() {
+      var result = [];
+      var categories = this.getElement().find('.organizer-category');
+      for (var i = 0; i < categories.count(); i++) {
+        result.push({
+          'id': categories.getItem(i).getAttribute('data-category-id'),
+          'name': categories.getItem(i).getParent().findOne('.organizer-category-name').$.value
+        })
+      }
+      return result;
+    },
+    getCategoryTerms: function() {
+      var result = [];
+      var categories = this.getElement().find('.organizer-category');
+      for (var i = 0; i < categories.count(); i++) {
+        var category = categories.getItem(i);
+        var terms = category.find('.organizer-term');
+        var termArray = [];
+        for (var j = 0; j < terms.count(); j++) {
+          termArray.push(terms.getItem(j).getAttribute('data-term-id'));
+        }
+        result.push({
+          'category': category.getAttribute('data-category-id'),
+          'terms': termArray
+        });
+      }
+      return result;
+    },
+    addTermToCategory: function(termName, categoryId) {
+      var _this = this;
+      var termObject = this.terms[termName];
+      if (termObject) {
+        var category = this.getElement().findOne('.organizer-category[data-category-id="' + categoryId + '"]');
+        var term = category.findOne('.organizer-term[data-term-id="' + termObject.id + '"]');
+        if (term != null) {
+          return;
+        }
+      }
+      else {
+        termObject = this.addTerm(this._generateTermId(), termName);
+      }
+      termObject.count++;
+      var termContainer = new CKEDITOR.dom.element('div');
+      termContainer.addClass('organizer-term-container');
+      var term = new CKEDITOR.dom.element('div');
+      term.addClass('organizer-term');
+      term.setAttribute('data-term-id', termObject.id);
+      term.setText(termName);
+      termContainer.append(term);
+      
+      var deleteTermLink = new CKEDITOR.dom.element('a');
+      deleteTermLink.addClass('icon-delete');
+      termContainer.append(deleteTermLink);
+      deleteTermLink.on('click', function() {
+        var termContainer = this.getParent();
+        var term = termContainer.findOne('.organizer-term');
+        _this.removeTerm(term.getAttribute('data-term-id'));
```

Comment:
Using _this is discouraged, use CKEditor.tools.bind instead (http://docs.ckeditor.com/#!/api/CKEDITOR.tools)